src/os/newstore/NewStore.cc

```diff
@@ -2783,6 +2783,8 @@ int NewStore::queue_transactions(
     _txc_add_transaction(txc, *p);
   }
 
+  reset_coll_object_tbl(tls);
```

Comment:
Because we must ensure that the transaction is not modified after we use coll_object_tbl to cache the encode result.
But https://github.com/ceph/ceph/blob/master/src/test/objectstore/store_test.cc#L1973https://github.com/ceph/ceph/blob/master/src/test/objectstore/store_test.cc#L1973,  the transaction is added new op (line 1981) after apply_transaction (line 1975) .
So I think when queue_transaction finish, we reset_coll_object_tbl is better.

Actually, coll_object_tbl is only useful for filestore, because newstore and keyvaluestore there is not encode transaction process. But in objectstore, we do not know which type of store backend.